### DISCUSSION TITLE
Do not allow download but only streaming

### DISCUSSION THREAD

**Discussion by DJ-SteffEN** [ampache/ampache#3890] [Q&A]
Hello

This may be a beginner's question, but how can I prevent direct downloads, e.g. by guests? 
For example through 'https://example.com/play/index.php?type=song&oid=4096&uid=-1&name=song.mp3' or similar urls

Thanks for your help
DJ Steffen

**Comment by lachlan-00** [comment]
most play calls require a valid session. are you trying to stop public access without a session?

**Comment by lachlan-00** [comment]
If you're trying to stop playback without a valid session in the website we could add some kind of cookie check for a browser before playback is allowed

**Comment by DJ-SteffEN** [comment]
Hi Lachlan,

thanks for your quick reply.
I also allow guests to stream, but not anyone to download (see [inthemix.at](https://inthemix.at)). It is unpleasant that anyone can still use it.
I think that guests will also need sessions, at least they can be seen in the Session table (User -1).
The player is called up via Javascript and the URL leads via .htaccess ... there must be a possibility.

Thanks in advance
DJ Steffen

**Comment by DJ-SteffEN** [comment]
Hi Lachlan,

I have solved it like this for now, it helps me for now.
I'm just wondering if there are other link combinations that can be used to get a download?
Do you know of any others? (except .../stream.php?action=download&xxx_id=x and the above URL)

.../src/Module/Applicatopn/Playback/PlayAction.php
-- line 1060 --
$headers = $this->browser->getDownloadHeaders($media_name, $mime, false, $stream_size);
-- line changed to --
$guid = (int)(Core::get_global('user')->id);
if (!$player && $guid == -1) {
    $headers = [];
    $headers['Content-Length'] = $stream_size;
} else {
    $headers = $this->browser->getDownloadHeaders($media_name, $mime, false, $stream_size);
}

Thanks in advance
DJ Steffen
